### PR TITLE
Update vmm-sys-util to 0.12.0 and prepare release 0.4.0

### PR DESCRIPTION
**Pull request by roypat**
### Summary of the PR

*Please summarize here why the changes in this PR are needed.*

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [ ] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [ ] All added/changed functionality has a corresponding unit/integration
  test.
- [ ] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [ ] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Upcoming version
+# Upcoming
```
Comment (JonathanWoollett-Light) [inline CHANGELOG.md:1]: Is there a reason for rename?

**Review by JonathanWoollett-Light — DISMISSED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (roypat) [reply CHANGELOG.md:1]: Just a miss on my part - I renamed the old heading to `v0.4.0` and then re-typed the "Upcoming" thing

Comment (roypat) [reply CHANGELOG.md]: Fixed (and also changed the overall formatting to match what the other rust-vmm crates use here)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,50 +1,54 @@
-# Upcoming version
+# Changelog
```
Comment (JonathanWoollett-Light) [inline CHANGELOG.md:1]: This should be `Changed`?

**Review by JonathanWoollett-Light — APPROVED** [review]

**Review by stsquad — APPROVED** [review]

**Review by jinankjain — APPROVED** [review]

**Review by stefano-garzarella — APPROVED** [review]

**Pull request merged**